### PR TITLE
buffered_test.go: Refactor create object, launch goroutine pattern

### PR DESCRIPTION
**Pull request by palash25**
Related to: https://github.com/kubernetes/kubernetes/issues/7923

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
None
```

### CODE REVIEW

File: staging/src/k8s.io/apiserver/plugin/pkg/audit/buffered/buffered_test.go [hunk 1]
```diff
@@ -173,7 +173,7 @@ func TestBufferedBackendShutdownWaitsForDelegatedCalls(t *testing.T) {
 
 	// Run backend, process events, wait for them to be batched and for delegated call to start.
 	stopCh := make(chan struct{})
-	backend.Run(stopCh)
+	go backend.Run(stopCh)
```
Comment (sttts) [inline staging/src/k8s.io/apiserver/plugin/pkg/audit/buffered/buffered_test.go]: Why? is that necessary?

Comment (palash25) [reply staging/src/k8s.io/apiserver/plugin/pkg/audit/buffered/buffered_test.go]: I thought that the [issue](https://github.com/kubernetes/kubernetes/issues/7923) required changing the pattern where an object was instantiated and a method defined on that object was run to be changed to a goroutine.

Comment (palash25) [reply staging/src/k8s.io/apiserver/plugin/pkg/audit/buffered/buffered_test.go]: it says 

> We should pick one and switch to it all over.

and the previous PR referenced on this issue changed the Run methods to be launched as goroutines.

Comment (mourya007) [inline staging/src/k8s.io/apiserver/plugin/pkg/audit/buffered/buffered_test.go]: `run` is a blocking method now. It waits for all listeners it started to finish. So i think we can use the goroutine for this.

**Pull request closed without merging**